lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -2715,6 +2717,46 @@ static T* castWithDebugInfo(SILInstruction *inst) {
   return cast<T>(inst);
 }
 
+/// Given an input type such as $Builtin.Int1, return the type decl of the
+/// corresponding stdlib numeric type, such as $Bool.
+/// See extractBuiltinTypeFromStdlibNumericType() on the reverse type
+/// conversion.
+static NominalTypeDecl *
+getStdlibNumericTypeDeclFromBuiltinType(Type ty, ASTContext &ctx) {
```

Comment:
Feels like this is the kind of helper method that the Swift codebase should already provide.

Reply:
Yeah, I think we have this somewhere in TFUtilities.cpp

Reply:
We have 2 functions in TFUtilities.cpp that are relevant:
1. `convertElementTypeToTensorValueType()` converts something like `Builtin.Int32` to `TensorHandle<Builtin.Int32>`
2. `convertSwiftTypeToTF()` converts something like `Builtin.Int32` or `$Int32` to a TF type like `TF_INT32`.

But I didn't find a function that converts `Builtin.Int32` to $Int32, so I wrote it here. It's based on the impl of `convertSwiftTypeToTF()`.

Another future use case of this new function is that I plan to use it to remove the `UncheckedRefCastInst` type of insts that graph lowering currently need to deal with (https://bugs.swift.org/browse/SR-8118).